src/Apps/Order/Routes/Shipping2/FulfillmentDetails.tsx

```diff
@@ -288,29 +273,40 @@ export const FulfillmentDetailsFragmentContainer = createFragmentContainer(
 interface LayoutProps
   extends Pick<
     FulfillmentDetailsFormProps,
-    | "order"
-    | "me"
-    | "setFulfillmentFormHelpers"
-    | "onSubmit"
-    | "active"
-    | "renderMissingShippingQuotesError"
+    "order" | "me"
+    // | "onSubmit"
   > {
   verifyAddressNow: boolean
-  addressVerificationComplete: () => void
+  addressVerificationComplete: () => Promise<void>
 }
 
 const FulfillmentDetailsFormLayout = (props: LayoutProps) => {
-  const { active } = props
+  const {
+    step,
+    savedOrderData: {
+      isArtsyShipping,
+      shippingQuotes,
+      lockShippingCountryTo,
+      fulfillmentType: savedFulfillmentType,
+      selectedSavedAddressId: savedSelectedAddressId,
+    },
+    helpers: {
+      fulfillmentDetails: { setFulfillmentFormHelpers },
+    },
+  } = useShippingContext()
```

Comment:
blocking: Avoid destructuring, especially at multiple levels deep. It is a trap, both in terms of understandability as well as safety: nested destructuring will blow up runtime code if some parent prop comes back null -- regardless of how it is typed. 

The above is a general comment, applied to javascript in general. But in terms of the shipping context -- just don't do it for the reasons discussed elsewhere 🙏

Reply:
Additionally, this kind of stuff:
```
fulfillmentType: savedFulfillmentType
```
or 
```
selectedSavedAddressId: savedSelectedAddressId,
```
is complexity. Why? because it is 1:2. You have variable a, which is named `fullfilmentType`. But now, we have `savedFulfillmentType`, aliased to `fulfilmentType`. If it is _actually_ `savedFulfillmentType`, then change the name in the context, and keep things 1:1. 

Aliasing in general should be avoided, as a rule. If there are variable clashes going on,  there's an elegant solution right at ones fingertips, which both reduces complexity  and increases comprehensibility of a system:

```tsx
const shippingContext = useShippingContext()
const foo = "hi"

if (foo !== shippingContext.foo) {
  return null
}
```